infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -313,6 +356,63 @@ private boolean runRecoveryReconfiguration() {
         return recoveryResult;
     }
 
+    /**
+     * Task to collect local node metrics.
+     * This task collects the following:
+     * Boolean Status of layout, sequencer and logunit servers.
+     * Logunit Metrics:
+     * Size of writeQueue for pending write requests.
+     * These metrics are then composed into NodeMetrics protobuf model
+     * and stored locally.
+     */
+    private void updateLocalMetrics() {
+        // Initializing the status of components
+        // No need to poll unless node is bootstrapped.
+        Layout layout = serverContext.getManagementLayout();
+        if (layout == null) {
+            return;
+        }
+
+        // Build and replace existing locally stored nodeMetrics
+        SequencerMetrics sequencerMetrics;
+        LogUnitMetrics logUnitMetrics;
+        if (layout.getSequencers().get(0).equals(getLocalEndpoint())) {
```

Comment:
you might want to update this code once the other PR is in. layout.getPrimarySequencer